src/ComposerConstraint.php

```diff
@@ -0,0 +1,90 @@
+<?php
+
+namespace Drupal\lightning_dev;
+
+class ComposerConstraint {
+
+  /**
+   * E.g. '^2.8 || ^3.0'.
+   *
+   * @var string
+   */
+  private $constraint = '';
+
+  /**
+   * @param string $constraint
+   */
+  public function __construct($constraint) {
+    $this->constraint = $constraint;
+  }
+
+  /**
+   * @return string[]
+   *   E.g. ['^2.8', '^3.0'].
+   */
+  public function getRanges() {
+    preg_match_all('/[0-9a-zA-Z\~\>\=\-\<\.\^\*]+/', $this->constraint, $matches);
+
+    return $matches[0];
+  }
+
+  /**
+   * @return string
+   *   E.g. '8.4.x-dev || 8.5.x-dev'.
+   */
+  public function getCoreDev() {
+    return $this->getDev([static::class, 'coreRangeToDev']);
+  }
+
+  /**
+   * @return string
+   *   E.g. '2.x-dev || 3.x-dev'.
+   */
+  public function getLightningDev() {
+    return $this->getDev([static::class, 'lightningRangeToDev']);
+  }
+
+  /**
+   * @param callable $callback
+   *
+   * @return string
+   */
+  private function getDev($callback) {
+    $dev = $this->constraint;
+    $ranges = $this->getRanges();
+
+    foreach ($ranges as $oldRange) {
+      $newRange = $callback($oldRange);
+      $dev = str_replace($oldRange, $newRange, $dev);
+    }
+
+    return $dev;
+  }
+
+  /**
+   * @param string $range
+   *   E.g. '8.5.3'.
+   * @return string
+   *   E.g. '8.5.x-dev'.
+   */
+  public static function coreRangeToDev($range) {
```

Comment:
Can these be private? If so, let's do that. We want to discourage people who aren't us from using this code :)

Reply:
Sure, I made it public because of the unit tests. Since it's enough to just test the interface, I'll make the other methods private too that are not used outside of the class.